### DISCUSSION TITLE
Is `Delegate.Method` intended to throw for non-public methods?

### DISCUSSION THREAD

**Discussion by DaZombieKiller** [dotnet/runtime#84560] [General] [answered]
`Delegate.Method` is currently [documented to throw](https://learn.microsoft.com/en-us/dotnet/api/system.delegate.method?view=net-7.0#exceptions) when the `MethodInfo` returned by it represents a method that the caller does not have access to. However, I have not been able to trigger this exception in any of the following tests under both .NET Core and .NET Framework:

* A delegate to a `private` method in a `public` class in the same assembly
* A delegate to a `private` method in a `public` class in another assembly
* A delegate to a `private` method in an `internal` class in another assembly using another `public` class as an accessor

Is this a documentation error? Or am I misunderstanding how this exception will normally be triggered? I would find it preferable for this property to not throw here, so the note in the documentation has left me somewhat concerned.

**Comment by jkotas** [comment, marked as answer]
The documentation is wrong. https://github.com/dotnet/dotnet-api-docs/pull/8956/ has the fix.
